src/Zynga/Framework/IO/Disk/V1/Manager.hh

```diff
@@ -0,0 +1,243 @@
+    if ($outFile === false) {
+      $this->fclose($inFile);
+      throw new FailedToOpenFileException($out);
+    }
+
+    while (!$this->feof($inFile)) {
+      $buffer = $this->fgets($inFile, 4096);
+      $this->bzwrite($outFile, $buffer, 4096);
+    }
+
+    if (!$this->fclose($inFile)) {
+      $this->bzclose($outFile);
+      throw new FailedToCloseFileException($in);
+    }
+
+    if (!$this->bzclose($outFile)) {
+      throw new FailedToCloseFileException($out);
+    }
+  }
+
+  // Have to use mixed for $handle here because Hack doesn't allow
+  // passing resource types to functions
+  protected function feof(mixed $handle): bool {
+    if (is_resource($handle)) {
+      return feof($handle);
+    }
+
+    return false;
+  }
+
+  // Have to use mixed for $handle here because Hack doesn't allow
+  // passing resource types to functions
+  protected function fgets(mixed $handle, int $maxBytesToRead): string {
+    if (is_resource($handle)) {
+      $result = fgets($handle, $maxBytesToRead);
+      if (is_string($result)) {
+        return $result;
+      }
+    }
+
+    return '';
+  }
+
+  protected function doesFileExist(string $fileName): bool {
+    return file_exists($fileName);
+  }
+
+  protected function fileOpen(string $fileName, string $mode): mixed {
+    try {
+      return fopen($fileName, $mode);
+    } catch (Exception $e) {
+      return false;
+    }
+  }
+
+  protected function directoryName(string $fileName): string {
+    return dirname($fileName);
+  }
+
+  protected function makeDirectory(string $path, int $permissions, bool $recursivePermissions): bool {
+    try {
+      return mkdir($path, $permissions, $recursivePermissions);
+    } catch (Exception $e) {
+      return false;
+    }
+  }
+
+  protected function unlink(string $fileName): bool {
+    try {
+      return unlink($fileName);
+    } catch (Exception $e) {
+      return false;
+    }
+  }
+
+  // Have to use mixed for $handle here because Hack doesn't allow
+  // passing resource types to functions
+  protected function fwrite(mixed $handle, string $dataToWrite): mixed {
+    if (is_resource($handle)) {
+      return fwrite($handle, $dataToWrite);
+    }
+
+    return false;
+  }
+
+  // Have to use mixed for $handle here because Hack doesn't allow
+  // passing resource types to functions
+  protected function fclose(mixed $handle): bool {
+    if (is_resource($handle)) {
+      return fclose($handle);
+    }
+
+    return false;
+  }
+
+  protected function isReadable(string $fileName): bool {
+    return is_readable($fileName);
+  }
+
+  protected function isWriteable(string $fileName): bool {
+    return is_writeable($fileName);
+  }
+
+  protected function bzopen(string $fileName, string $mode): mixed {
+    try {
+      return bzopen($fileName, $mode);
+    } catch (Exception $e) {
+      return false;
+    }
+  }
+
+  // Have to use mixed for $handle here because Hack doesn't allow
+  // passing resource types to functions
```

Comment:
Looking at the docs, I think that's not true. I think you can change this to use 'resource' as the primitive and it should work. 

```
https://docs.hhvm.com/hack/types/type-system

<?hh // partial

class fizz {
  public static function foo(resource $fizz):void { // this works
      fclose($fizz);
  }
}

$tmpfile = tmpfile();
fizz::foo($tmpfile);